//
// PROJECT:         Aspia
// FILE:            category/category_power_options.proto
// LICENSE:         GNU Lesser General Public License 2.1
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto;

message PowerOptions
{
    enum PowerSource
    {
        POWER_SOURCE_UNKNOWN    = 0;
        POWER_SOURCE_DC_BATTERY = 1;
        POWER_SOURCE_AC_LINE    = 2;
    }

    enum BatteryStatus
    {
        BATTERY_STATUS_UNKNOWN    = 0;
        BATTERY_STATUS_HIGH       = 1;
        BATTERY_STATUS_LOW        = 2;
        BATTERY_STATUS_CRITICAL   = 3;
        BATTERY_STATUS_CHARGING   = 4;
        BATTERY_STATUS_NO_BATTERY = 5;
    }

    message Battery
    {
        enum State
        {
            STATE_UNKNOWN      = 0;
            STATE_CHARGING     = 1;
            STATE_CRITICAL     = 2;
            STATE_DISCHARGING  = 4;
            STATE_POWER_ONLINE = 8;
        }

        enum Type
        {
            TYPE_UNKNOWN = 0;
            TYPE_PBAC    = 1;
            TYPE_LION    = 2;
            TYPE_NICD    = 3;
            TYPE_NIMH    = 4;
            TYPE_NIZN    = 5;
            TYPE_RAM     = 6;
        }

        string device_name          = 1;
        string manufacturer         = 2;
        string manufacture_date     = 3;
        string unique_id            = 4;
        string serial_number        = 5;
        string temperature          = 6;
        int32 design_capacity       = 7;
        Type type                   = 8;
        int32 full_charged_capacity = 9;
        int32 depreciation          = 10;
        int32 current_capacity      = 11;
        int32 voltage               = 12;
        uint32 state                = 13;
    }

    PowerSource power_source          = 1;
    BatteryStatus battery_status      = 2;
    int64 full_battery_life_time      = 3;
    int64 remaining_battery_life_time = 4;
    int32 battery_life_percent        = 5;
    repeated Battery battery          = 6;
}
